File: web-sdk/src/data/services/yalo-message/yalo-message-auth-service.ts
// Copyright (c) Yalochat, Inc. All rights reserved.

import type { Result } from '@domain/common/result';
import type { AuthResponse } from '@domain/models/events/external_channel/in_app/sdk/sdk_message';

export type { AuthResponse };

export interface YaloMessageAuthService {
  fetchToken(): Promise<Result<AuthResponse>>;
  refreshToken(refreshToken: string): Promise<Result<AuthResponse>>;
}

File: android-sdk/sdk/src/main/java/ai/yalo/chat/sdk/data/services/auth/YaloMessageAuthService.kt
// Copyright (c) Yalochat, Inc. All rights reserved.
package ai.yalo.chat.sdk.data.services.auth

/**
 * The two calls the backend offers for getting into a conversation.
 *
 * Nothing here remembers anything. Deciding when a token is worth asking for,
 * and what to do with one afterwards, belongs to whoever calls this.
 */
internal interface YaloMessageAuthService {

    suspend fun fetchToken(): Result<AuthCredentials>

    suspend fun refreshToken(refreshToken: String): Result<AuthCredentials>
}

/** The lifetime arrives as a length, so it only means something next to a clock. */
internal data class AuthCredentials(
    val accessToken: String,
    val refreshToken: String,
    val expiresInSeconds: Long,
)
